webform_civicrm.module

```diff
@@ -499,10 +500,52 @@ function webform_civicrm_civicrm_merge($type, $data, $new_id = NULL, $old_id = N
       ->condition('contact_id', '%-' . $old_id . '-%', 'LIKE')
       ->execute();
     // Update contact reference field data
-    db_query("UPDATE {webform_submitted_data} d, {webform_component} c SET d.data = :new
-      WHERE d.data = :old AND d.cid = c.cid AND d.nid = c.nid AND c.type = 'civicrm_contact'",
-      array(':new' => $new_id, ':old' => $old_id)
-    );
+    $webforms = \Drupal\webform\Entity\Webform::loadMultiple();
+    foreach ($webforms as $webform) {
+      $elements = $webform->getElementsInitializedAndFlattened();
+      foreach ($elements as $key => $value) {
+        $handler_collection = $webform->getHandlers('webform_civicrm');
+        if (!$handler_collection->has('webform_civicrm')) {
+          continue;
+        }
+        $submissions = \Drupal::entityTypeManager()
+            ->getListBuilder('webform_submission')
+            ->getStorage()
+            ->loadByProperties([
+                'webform_id' => $webform->id(),
+            ]);
+        foreach ($submissions as $submission) {
+          $sid = $submission->get('sid')->value;
+          //Update old Drupal User ID with new Drupal User ID
+          /** @var Drupal\webform\Entity\WebformSubmission $submission */
+          $submission = \Drupal\webform\Entity\WebformSubmission::load($sid);
+          $suid = $submission->getOwnerId();
+
+          \Drupal::service('civicrm')->initialize();
```

Comment:
Let's run this at the beginning instead of constantly. It is reentrant but it's burning unnecessary cycles.